__main__.py

```diff
@@ -3,81 +3,149 @@
+    WITHDRAWN = "Withdrawn"
+    SUPERSEDED = "Superseded"
+
 
 COLOR_CODES = {
-    "Accepted": "&2",
-    "Active": "&a",
-    "Deferred": "&6",
-    "Draft": "&8",
-    "Final": "&1",
-    "Provisional": "&5",
-    "Rejected": "&4",
-    "Replaced": "&e",
-    "Withdrawn": "&d",
+    Status.ACCEPTED: "&2",
+    Status.ACTIVE: "&a",
+    Status.DEFERRED: "&6",
+    Status.DRAFT: "&8",
+    Status.FINAL: "&1",
+    Status.PROVISIONAL: "&5",
+    Status.REJECTED: "&4",
+    Status.REPLACED: "&e",
+    Status.WITHDRAWN: "&d",
+    Status.SUPERSEDED: "",  # FIXME Pick color
 }
-LATEST = Path("pep-tracker/latest.json")
+
+LATEST = Path(__file__).parent.resolve() / "latest.json"
 URL = "https://peps.python.org/api/peps.json"
+PEP = NewType("PEP", str)
+
+
+def now() -> dt.datetime:
+    return dt.datetime.now(tz=dt.timezone.utc)
+
+
+class StateDict(TypedDict):
+    version: str
+    data: dict[str, str]
+    time: str
 
 
 class State:
+    version: ClassVar[str] = __version__
+
     def __init__(
-        self, data: dict[str, str], registered_at: dt.datetime | None = None
+        self, data: dict[PEP, Status], time: dt.datetime | None = None
     ) -> None:
         self._data = data
-        self.date = registered_at or dt.datetime.now(tz=dt.timezone.utc)
+        self.time = time or now()
 
-    def __getitem__(self, key: str) -> str:
+    def __getitem__(self, key: PEP) -> Status:
         return self._data[key]
 
-    def __xor__(self, other: State) -> dict[str, tuple[str, str]]:
-        return {k: (self[k], other[k]) for k in self._data if self[k] != other[k]}
+    def __xor__(self, other: State) -> dict[PEP, tuple[Status, Status]]:
+        # FIXME Handle PEPs that have been newly added, and are only on `other`
+        return {k: (self[k], other[k]) for k in self._data if self[k] is not other[k]}
+
+    @classmethod
+    def from_api(cls, data: dict[str, dict[str, str]], /) -> Self:
+        return cls(data={PEP(k): Status(v["status"]) for k, v in data.items()})
+
+    @classmethod
+    def _migrate(cls, old: dict[str, Any] | StateDict, /) -> StateDict:
+        out = StateDict(version=cls.version, data={}, time=now().isoformat())
+        old_version = old.get("version")
+        if old_version is None:
+            data = cast("dict[str, str]", old.copy())
+            out["data"] = data
+        else:
+            out = cast(StateDict, old)
+        return out
+
+    @classmethod
+    def from_dict(cls, state: StateDict | dict[str, Any], /) -> Self:
+        state_dict = cls._migrate(state)
+        data_raw = state_dict["data"]
+        data = {PEP(k): Status(v) for k, v in data_raw.items()}
+        time = dt.datetime.fromisoformat(state_dict["time"])
+        return cls(data=data, time=time)
+
+    def to_dict(self) -> StateDict:
+        return StateDict(
+            version=self.version,
+            data={str(k): v.value for k, v in self._data.items()},
+            time=self.time.isoformat(),
+        )
```

Comment:
Converted some module functions to methods of this class. I believe more functions could be turned into methods, but I'm not sure if that'd be desirable.